### ISSUE TITLE
Mouse emulation suggestion

### ISSUE THREAD

**Issue by INemesisI** [opentrack/opentrack#113]
Hey,
i would like to use the mouse emulation to translate movements of my phone (FreePie) into FPS games (for a cheap oculus rift). Most FPS games calculate the dx and dy values of the mouse relative to the center of the screen and set the position of the mouse back to the center. 
Since the emulation software then sets the position of the mouse to a absolute position again, those dx and dy values get very big and the view keeps rotating.
Would it be possible to add a dynamic mode, where the software only changes the relative position of the mouse, rather than setting it to an absolute position on the screen?

**Comment by FlyingCircus-** [comment]
I'm not sure if what you're describing is just the motion drift inherent to the innertial trackers. Accelerometers aren't completelly accurate and work in a relative basis, as you're suggesting. I believe you need to set up a centering key (through the shortcuts menu included in OpenTrack) to re-center the view when the drift is becoming too high.

**Comment by mm0zct** [comment]
I think I agree more with INemesisI's interpretation of the situation. With a gyro based tracker you get absolute rotation drift over time, so forward is no longer exactly forward, but if you stand still the view will always be still (to within the noise of the sensor and you're ability to actually remain stationary).

I noticed this behaviour with Freetrack before but only really used Opentrack for TrackIR  based games, so didn't have much incentive to fix it.

Last time I tried it OpenTrack's mouse emulation does work by setting the absolute position out the mouse output. This fights with most FPS games mouse handling which works by centering the mouse every frame and getting the difference in location at the start of the frame to compute a mouse delta. Since OpenTrack keeps putting the mouse back where it things it should be every tracker update this leads to the game and OpenTrack fighting over the mouse, and the game spinning the world around nauseatingly.

The solution if to query the mouse location before setting the new position, applying the translation delta to the updated location rather than the last location OpenTrack expected it to be.

I'll maybe look into it this week.

**Comment by FlyingCircus-** [comment]
Very clear explanation, I see the problem now. Ignore my previous input.

**Comment by sthalik** [comment]
We should use angles 0->180 converted to integers of 0->resolution for keeping deltas. Floats shouldn't be used. Queried mouse position should only be used for updating it.

TL;DR mouse shouldn't drift from center due to its imprecision.

@mm0zct cool!

**Comment by thmundal** [comment]
When will we see a build including this fix? Is there any where to get the version with this implemented?

**Comment by sthalik** [comment]
@thmundal next RC will have a fix, tested with Witcher 3 and New Vegas. It should eliminate cursor drift too.
